### DISCUSSION TITLE
Use MapLayersWidgetControl to report an error

### DISCUSSION THREAD

**Discussion by shenshengyi** [iTwin/itwinjs-core#1348] [Q&A] [answered]
hi，When I use MapLayersWidgetControl, an error always appears, prompting ：Thanks.
![图片](https://user-images.githubusercontent.com/26344907/117308029-b391fc00-aeb3-11eb-9675-621f736de82a.png)

**Comment by calebmshafer** [comment, marked as answer]
@shenshengyi how are you including the `MapLayersWidgetControl` into your application? The `MapLayerSettingsService.onCustomLayerNameRemoved` looks to be a static property that is initialized by default so not quite sure why it'd be undefined.

**Reply by shenshengyi** [reply]
Hi.
I added "@bentley/map-layers": "^2.10.0", or "@bentley/map-layers": "^2.12.0" to package.json, and when I use npm install, just run the program Will report this error.

This error disappeared when I used npm i @bentley/map-layers to install the dependencies. 

first:
    await MapLayersUI.initialize(false, IModelApp.i18n);

than:

![图片](https://user-images.githubusercontent.com/26344907/117471441-6f255f80-af8a-11eb-8270-bc721e641758.png)

I found that there is no such field in the generated *.d.ts file. 


![图片](https://user-images.githubusercontent.com/26344907/117472059-14d8ce80-af8b-11eb-837e-e378e27a2805.png)

**Reply by pmconne** [reply]
@shenshengyi you should check what version of the package actually got installed. The onCustomLayerNameRemoved property was introduced long after 2.12.0 - most likely you had installed a version from before that property was introduced.

**Reply by shenshengyi** [reply]
Thank you very much, I understand, I did fail in version 2.10. Let me upgrade the version.
